sql/sql_statistics.h

```diff
@@ -326,9 +339,54 @@ class Histogram : public Histogram_base
   /*
     Estimate selectivity of "col=const" using a histogram
   */
-  double point_selectivity(double pos, double avg_sel);
+  double point_selectivity(double pos, double avg_sel) override;
 };
 
+class Histogram_json : public Histogram_base
+{
+private:
+  Histogram_type type;
+  uint8 size; /* Number of elements in the histogram*/
+  uchar *values;
+
+public:
+  bool parse(MEM_ROOT *mem_root, Histogram_type type_arg, const uchar *ptr, uint size) override {return false;}
+
+  void serialize(Field *to_field) override{}
+
+  uint get_size() override {return (uint) size;}
```

Comment:
What does this function do?  Number of buckets is returned by `get_width` (see below). Please check if we can remove it.

Reply:
I think it's used currently to fill the `hist_size` column for binary histograms. We can use `get_width` for that, but I think it gets halved for double-precision-binary-histograms.